### DISCUSSION TITLE
Timout after (long) time of no interaction

### DISCUSSION THREAD

**Discussion by klubdergutenseiten-user** [SafeExamBrowser/seb-win-refactoring#675] [Q&A]
Hi there
We experienced a situation where we are not sure if this intended or not. Let's say you are in the SEB and running an exam where you have a long period of no interaction with the computer (example: you start your exam with SEB, then do some exercises of the screen, return to your exam). During this time the exam-software (client side) regularly sends a heartbeat via ajax to the exam-server. Now we noticed, that after some undefined time (around 20-40 Minutes) the heartbeat fails and the SEB stops sending those ajax heartbeats. The SEB is still running and at the client side no error occures.

Now: is this intended? How can we prevent from this happening (without reloading the exam-site regularly on no interaction)? Or how can we further investigate this to help you / us find a solution?

Many thanks and regards,
florian

**Comment by dbuechel** [comment]
I doubt that it is (at least definitely not from our side), but maybe Chromium resp. CEF has some functionality which pauses the execution of (certain) JavaScript code after a period of inactivity; i.e. the pausing may be caused by the way you're implementing the heartbeat. Another possibility could be an energy saving option which is active on a particular device (type) and thus causing the problem.

Either way, you can enable the live application log (see Security > Application Log) to see what's going on in the inner workings of SEB, and you can furthermore access the browser log file for a session under `%LocalAppData%\SafeExamBrowser\Logs`.
